### ISSUE TITLE
Memory leak when querying from not null column tables

### ISSUE THREAD

**Issue by zzcclp** [Kyligence/ClickHouse#52]
> You have to provide the following information whenever possible.

Memory leak when querying from not null column tables.

原先的 MergeTree 数据是以 not null 创建的，在支持了 nullable 后，则建表语句中的列都加了 not null，执行 TPCH 100 时，执行到 Q17 就把 128 G 内存都吃光了，但查询加了 Nullable(type) 的表，执行完 21 条 SQL 内存使用大概将近 60 G。

Root cause:
在使用 not null 列表后，某些 Filter 只执行 isnotnull 的 filter 就不再需要，比如下面两张图的对比：
![8l2or6PhCj](https://user-images.githubusercontent.com/9430290/185603513-d2e04b47-2ec6-41df-917f-f5895ebd969c.png)
![WjxrGpMn2E](https://user-images.githubusercontent.com/9430290/185603530-ad8ecbf7-1650-430a-aa45-e6bd970dbb80.png)

此时，FileScan 又是拆分后的，执行后直接对接 Exchange，这时因为 CloseableIterator 都是在 WholeStageTransformer 中加的，这种场景下没有 WholeStageTransformer，因此没有 CloseableIterator，导致内存泄漏。

在 #43 中解决。

> A clear and concise description of what works not as it is supposed to.

**Does it reproduce on recent release?**

[The list of releases](https://github.com/ClickHouse/ClickHouse/blob/master/utils/list-versions/version_date.tsv)

**Enable crash reporting**

> If possible, change "enabled" to true in "send_crash_reports" section in `config.xml`:

```
<send_crash_reports>
        <!-- Changing <enabled> to true allows sending crash reports to -->
        <!-- the ClickHouse core developers team via Sentry https://sentry.io -->
        <enabled>false</enabled>
```

**How to reproduce**

* Which ClickHouse server version to use
* Which interface to use, if matters
* Non-default settings, if any
* `CREATE TABLE` statements for all tables involved
* Sample data for all these tables, use [clickhouse-obfuscator](https://github.com/ClickHouse/ClickHouse/blob/master/programs/obfuscator/Obfuscator.cpp#L42-L80) if necessary
* Queries to run that lead to unexpected result

**Expected behavior**

> A clear and concise description of what you expected to happen.

**Error message and/or stacktrace**

> If applicable, add screenshots to help explain your problem.

**Additional context**

> Add any other context about the problem here.

**Comment by zzcclp** [comment]
Fixed in #43

**Comment by baibaichen** [comment]
> Fixed in #43 

which PR?

**Comment by zzcclp** [comment]
> > Fixed in #43
> 
> which PR?

https://github.com/oap-project/gluten/pull/326
